### PR TITLE
update ui of overview screen to include location sharing toggle

### PR DESCRIPTION
**Pull request by marinovdeyan**
This PR includes UI changes that go with the backend modifications made in PR #253.

We are now able to switch on and off the location sharing with a friend through a toggle on the overview screen. This toggle has a direct impact on if this friend can see us on the map or not.

We now see only the friends that shared their location with us on the map.

I also added some simple UI tests to accompany this new modification.

### CODE REVIEW

File: app/src/main/java/com/swent/suddenbump/model/user/UserViewModel.kt [hunk 1]
```diff
@@ -211,6 +211,7 @@ open class UserViewModel(
         uid,
         onSuccess = {
           _user.value = it
+          Log.d(logTag, "User set 2: ${_user.value}")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/model/user/UserViewModel.kt:214]: Maybe you can delete this log.

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt [hunk 2]
```diff
@@ -61,7 +63,11 @@ fun OverviewScreen(navigationActions: NavigationActions, userViewModel: UserView
   val groupedFriends by userViewModel.groupedFriends.collectAsState()
 
   // Charge les emplacements des amis lorsque l'écran est composé
-  LaunchedEffect(Unit) { userViewModel.loadFriends() }
+  LaunchedEffect(Unit) {
+    Log.d("OverviewScreen", "currentUser: $currentUser")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt:67]: Maybe you can remove this

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt [hunk 3]
```diff
@@ -61,7 +63,11 @@ fun OverviewScreen(navigationActions: NavigationActions, userViewModel: UserView
...
+    Log.d("OverviewScreen", "groupedFriends: $groupedFriends")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt:68]: Maybe you can remove this log as it's not useful

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt [hunk 4]
```diff
@@ -117,6 +123,7 @@ fun OverviewScreen(navigationActions: NavigationActions, userViewModel: UserView
                     .padding(horizontal = 16.dp),
             horizontalAlignment = Alignment.CenterHorizontally) {
               if (groupedFriends?.isNotEmpty() == true) {
+                Log.d("OverviewScreen", "groupedFriends 2: $groupedFriends")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt:126]: Maybe you can remove this log

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt [hunk 5]
```diff
@@ -170,10 +177,18 @@ fun CategoryHeader(category: DistanceCategory) {
...
 @Composable
 fun UserRow(user: User, navigationActions: NavigationActions, userViewModel: UserViewModel) {
   val coroutineScope = rememberCoroutineScope()
   var locationText by remember { mutableStateOf("Loading...") }
+  val locationSharedWith by userViewModel.locationSharedWith.collectAsState()
+  var isLocationShared = false
+  locationSharedWith.forEach { friend ->
+    isLocationShared = isLocationShared || friend.uid == user.uid
+  }
+  Log.d("UserRow", "isLocationShared: $isLocationShared")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt:190]: Maybe remove this

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt [hunk 6]
```diff
@@ -170,10 +177,18 @@ fun CategoryHeader(category: DistanceCategory) {
...
+  Log.d("UserRow", "locationSharedWith: $locationSharedWith")
```
Comment (ifaresse) [inline app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt:191]: Maybe remove this

**Review by ifaresse — APPROVED** [review]
photo_2024-12-06_03-50-02
The overall changes on the overview looks good and the location switch are working !
LGTM just act upon the comments before merging.

File: app/src/main/java/com/swent/suddenbump/model/user/UserViewModel.kt (hunk 1 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/model/user/UserViewModel.kt]: done

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt (hunk 2 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt]: done

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt (hunk 4 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt]: done

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt (hunk 6 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt]: done

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt (hunk 5 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt]: done

File: app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt (hunk 3 above)
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/ui/overview/Overview.kt]: done

**Comment by marinovdeyan** [comment]
How come the line coverage on new code decreased by 5% between commits eb90f8a and eef0cd5 when all I did was delete the logs?? We need to have a look at the way SonarCloud calculates the coverage because this doesn't seem right.

**Pull request merged**